### ISSUE TITLE
Slow performance with camera

### ISSUE THREAD

**Issue by NBatalha96** [jfeinstein10/SlidingMenu#649]
Hi!

Well, I'm making my final school project and the project evolves the ability to read qr codes.
So the problem is when the camera is on, the performance of sliding the menu is really bad. Am I implementing the camera bad? I don't think so. So to solve the problem I've tried to turn off the camera when I slide the fragment, but had no success (didn't find any method from SlidingMenu). I found the setOnOpenListener. But that is not what I want.

Basically the performance of sliding the menu is slow when the camera is on. To fix that I thought to deactivate the camera after move one pixel when opening the SlidingMenu. That would just give me the performance I've been looking for.

Can anyone help me?
If I wasn't clear, please tell me, I really need to get this working.

**Comment by poliver** [comment]
Take a look at setOnCloseListener() and setOnClosedListener() in SlidingMenu. Similar methods are available for opening the menu.

**Comment by NBatalha96** [comment]
Well, it's something like that. But that triggers when the SlidingMenu closes. And I want before that and when opening the SlidingMenu, not when closing. I want a method that triggers when I try to open the SlidingMenu, while I'm opening, not when it's open.

**Comment by poliver** [comment]
Yea, so use setOnOpenListener

**Comment by NBatalha96** [comment]
But that is triggered when the SlidingMenu opens. Not when I slide it 1 pixel for example. I want while opening, not when opened.

**Comment by poliver** [comment]
There is a difference between onOpenListener and onOpenedListener. Seems like you could use both to accomplish what you need.

**Comment by NBatalha96** [comment]
I think you are not understanding what I want.

onOpenedListener:
OnOpenedListener.onOpened() will be called after the SlidingMenu is opened.

onOpenListener:
OnOpenListener.onOpen() will be called when the SlidingMenu is opened.

They are triggered after or when the SlidingMenu is open. I want before I slide it all. I will give an example to make you understand.
You try to open the SlidingMenu, and when you move just 1 pixel (of the SlidingMenu) it triggers something. So the OnOpenListener and OnOpenedListener are triggered when the SlidingMenu is all open, not when it moves just a pixel.

**Comment by poliver** [comment]
Oh, I see what you mean. I added the functionality that you need to my fork. Currently, it doesn't have a way to detect which menu you are working with, but it does allow you to add callbacks for whenever you initiate an open or a close, and it should give you a general idea.

https://github.com/phil-oliver/SlidingMenu
